docs/proposals/resource-catalog.md

```diff
@@ -0,0 +1,222 @@
+
+
+We can reshape the catalog resource to more easily express them, secure them, and claim them.
+
+# Goals
+ 1.  cluster-admin should control which catalog exist.
+ 2.  cluster-admin should be able/disable per-catalog exposure on a per-project granularity.
+ 3.  project-admin should be able to delegate per-catalog exposure.
+ 4.  project-editor should be able to claim an entry from the catalog.
+ 5.  ResourceCatalogClaims should be sync-able, but not require syncing.
+ 6.  project-admin should be able to control the shape of claim realization and prevent unwanted injection.
+ 7.  ResourceCatalogEntries should not be restricted to specific claim realization mechanism.
+
+To achieve the ACL related goals, we can create namespaced resources that are selected by label at a cluster level.
+
+
+# API Objects
+
+## ResourceCatalog
+ResourceCatalog is a cluster-scoped resource.  It has a cluster-admin controlled spec that is used to drive a controller.
+The spec includes a label selector that is used in a list/watch to filter all ResourceCatalogEntries in the cluster.
+The controller will find matching entries and copy a reference into the ResourceCatalog.Status field.  The controller
+should have a plug-point for deciding if a given entry is accepted into a given catalog: `Accept(catalogName, entryNamespace, entryName)`
+is probably sufficient.
+
+```go
+// ResourceCatalog is a cluster-scoped resource with a specified spec and a controller maintained status
+type ResourceCatalog struct {
+	unversioned.TypeMeta
+	api.ObjectMeta
+
+	Spec   ResourceCatalogSpec
+	Status ResourceCatalogStatus
+}
+
+type ResourceCatalogSpec struct {
+	// Selector is the selector to use when gathering the namespace scope `ResourceCatalogEntries` for
+	// the `Status`
+	Selector map[string]string
```

Comment:
> It sounds like for the MVP we're saying if you are allowed to creating a posting, that's the only access check we need for starters.

I really think we should consider the case of multiple catalogs, where one is "blessed" and another is not.  However, I will admit that if you don't have that distinction you can still create something.

Even if you pin the label selector to only checking a single label, I can see it as a useful pattern and extends our normal resource selection rules.